### PR TITLE
Issue 4387 gl tf

### PR DESCRIPTION
**Pull request by Brijendra21**
Pull request overview
---------------------

Enhancement : OpenStudio could export a web standard GLTF

### Pull Request Author

 - [ ] Model API Changes / Additions
 - [ ] Any new or modified fields have been implemented in the EnergyPlus ForwardTranslator (and ReverseTranslator as appropriate)
 - [ ] Model API methods are tested (in `src/model/test`)
 - [ ] EnergyPlus ForwardTranslator Tests (in `src/energyplus/Test`)
 - [ ] If a new object or method, added a test in NREL/OpenStudio-resources: Add Link
 - [ ] If needed, added VersionTranslation rules for the objects (`src/osversion/VersionTranslator.cpp`)
 - [ ] Verified that C# bindings built fine on Windows, partial classes used as needed, etc.
 - [ ] All new and existing tests passes
 - [ ] If methods have been deprecated, update rest of code to use the new methods

**Labels:**

 - [ ] If change to an IDD file, add the label `IDDChange`
 - [ ] If breaking existing API, add the label `APIChange`
 - [ ] If deemed ready, add label `Pull Request - Ready for CI` so that CI builds your PR

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] Code Style, strip trailing whitespace, etc.
 - [ ] All related changes have been implemented: model changes, model tests, FT changes, FT tests, VersionTranslation, OS App
 - [ ] Labeling is ok
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified

### CODE REVIEW

File: src/utilities/core/Enum.hpp
Comment (Brijendra21) [reply src/utilities/core/Enum.hpp:75]: I'm assuming Clang introduced that change..

File: src/model/test/GltfForwardTranslator_GTest.cpp
Comment (Brijendra21) [reply src/model/test/GltfForwardTranslator_GTest.cpp:2]: Done

File: src/model/test/GltfForwardTranslator_GTest.cpp
Comment (Brijendra21) [reply src/model/test/GltfForwardTranslator_GTest.cpp:46]: Done

File: src/model/test/GltfForwardTranslator_GTest.cpp
Comment (Brijendra21) [reply src/model/test/GltfForwardTranslator_GTest.cpp:53]: Removed /// visual studio type comments and comment tags to plain and simple //.
Looking into the latter part...

File: src/model/GltfForwardTranslator.hpp
Comment (Brijendra21) [reply src/model/GltfForwardTranslator.hpp:82]: Added..

File: src/model/GltfForwardTranslator.hpp
Comment (Brijendra21) [reply src/model/GltfForwardTranslator.hpp:2]: Done..

File: src/model/GltfForwardTranslator.cpp
Comment (Brijendra21) [reply src/model/GltfForwardTranslator.cpp:2]: Done

File: src/model/GltfForwardTranslator.cpp
Comment (Brijendra21) [reply src/model/GltfForwardTranslator.cpp:92]: took care of it..

File: src/model/GltfForwardTranslator.cpp
Comment (Brijendra21) [reply src/model/GltfForwardTranslator.cpp:126]: done.

File: src/model/GltfForwardTranslator.cpp
Comment (Brijendra21) [reply src/model/GltfForwardTranslator.cpp:257]: done

File: src/model/GltfForwardTranslator.hpp
Comment (Brijendra21) [reply src/model/GltfForwardTranslator.hpp]: made changes accordingly.

File: src/model/test/GltfForwardTranslator_GTest.cpp
Comment (Brijendra21) [reply src/model/test/GltfForwardTranslator_GTest.cpp]: C++ gltf validator's status is still experimental (W.I.P). Almost every rendering platform provides the validation results as soon as we try rendering a gltf so I think this isn't an issue.

File: src/model/GltfForwardTranslator.cpp
Comment (jmarrec) [inline src/model/GltfForwardTranslator.cpp:718]: What's the point of this divider = 1?

**Review by jmarrec — CHANGES_REQUESTED** [review]
There are portion of the code that I'm not qualified/smart enough to fully understand but there are a few C++ improvements and code smell I'm seeing.

As this is a rather big amount of review comments, I'll going to a make a PR with some refactoring of this, and I'll let you take a look at it before you merge (partially or fully) into this branch.

File: src/model/GltfForwardTranslator.cpp
Comment (Brijendra21) [reply src/model/GltfForwardTranslator.cpp:568]: // Adds & Creates Face Indices buffers and Accessors
  // This method infers the indicesbuffer and adds them to the glTF model accessor
  // which expects two types of accessor input one from indicesbuffer and the second
  // one form coordinates buffer. So here after appending the buffer content it is sending
  // over the index so the containing node will be aware of which one to refer.
  // A better overview here at [Infographic  ](https://github.com/KhronosGroup/glTF/blob/main/specification/2.0/figures/gltfOverview-2.0.0b.png)
  // param : faceIndices
  // param : _indicesBuffer
  // param : _accessors
  // returns : index of the Face Indices

File: src/model/GltfForwardTranslator.cpp
Comment (Brijendra21) [reply src/model/GltfForwardTranslator.cpp:1033]: if I'm changing then top node is not set properly and rendered orientation isn't correct.
image

Originally 
image

Comment (jmarrec) [reply src/model/GltfForwardTranslator.cpp]: @Brijendra21 Couple of things then:
* This isn't captured in unit tests and probably should (all tests were passing locally when I made my changes)
* we should identify why and correct it, it's probably just a small hiccup. It's a waste to have to create a new vector and push back all elements into it. Just ensure topNode is the first one to begin with.

**Pull request merged**
